Source/Core/TrackedNodeComponent.cs

```diff
@@ -2,7 +2,10 @@
 
 namespace Microsoft.Boogie
 {
-  public abstract record ProgramElement()
+  // Represents an AST node, or component of a node, that is being
+  // tracked during the proof process to determine whether it was
+  // used as part of a completed proof.
+  public abstract record TrackedNodeComponent()
```

Comment:
I think `///` would be better here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/

If you then use Rider to move `TrackedNodeComponent`, the comment will be moved with it, while the `//` is AST node wise not tied to `TrackedNodeComponent`.